File: patterns/01-design-patterns/03-behavioral/07-mediator/python/mediator.py
class PatternError(Exception):
    def __init__(self,code):super().__init__(code);self.code=code
class SupportMediator:
    routes={"customer":["agent","bot"],"agent":["customer"],"bot":["agent"]}
    def dispatch(self,event):
        recipients=self.routes.get(event.get("from"))
        if recipients is None:raise PatternError("unsupported_sender")
        return[{"from":event["from"],"to":target,"message":event["message"]} for target in recipients]
def evaluate(input_data):
    mediator=SupportMediator();deliveries=[]
    for event in input_data.get("events",[]):deliveries.extend(mediator.dispatch(event))
    return{"deliveries":deliveries}

File: patterns/01-design-patterns/03-behavioral/07-mediator/js/mediator.mjs
export class PatternError extends Error{constructor(code){super(code);this.code=code;}}
class SupportMediator{routes={customer:["agent","bot"],agent:["customer"],bot:["agent"]};dispatch(event){const recipients=this.routes[event.from];if(recipients===undefined)throw new PatternError("unsupported_sender");return recipients.map((target)=>({from:event.from,to:target,message:event.message}));}}
export function evaluate(input){const mediator=new SupportMediator();return{deliveries:(input.events??[]).flatMap((event)=>mediator.dispatch(event))};}
